### PR TITLE
Added C++ codes under the language section 

### LINKED ISSUE #589: [Feature Request] : Add C++ under languages section
### Is your feature request related to a problem? Please describe.
I want to add C++ documentation under the language section .

### Describe the solution you'd like
The C++ documentation will include markdown files explaining key concepts i.e. Introduction , Variables , Operators etc..

**Would you like to work on this feature?**  
- [X] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by haseebzaki-07**
### Description
Added C++ introduction and functions codes in the CPP folder under the languages section.
Fixes #589 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/languages/cpp/cp-0.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+```cpp
+// This is a single-line comment
+/* This is a 
+   multi-line comment */
+```
+
+
+Semicolons: Each statement in C++ ends with a semicolon (;).
+
+Braces: Curly braces ({}) are used to define the beginning and end of code blocks.
```
Comment (ajay-dhangar) [inline docs/languages/cpp/cp-0.md]: replace `({})` with ``` (`{}`) ```

File: docs/languages/cpp/cp-5.md
Comment (ajay-dhangar) [inline docs/languages/cpp/cp-5.md:1]: I think the existing content in `docs/languages/cpp/cp-5.md` is good. I don't think your code is better, so I don't think it should be edited into the file.

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by haseebzaki-07** [comment]
hey @ajay-dhangar I made the changes

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**